Ingest script

mvandenburgh opened the pull request.

mcovalt commented on django/src/rdwatch/management/commands/ingest.py:
```diff
@@ -0,0 +1,272 @@
+        else:
+            assert sensor_name
+
+            sensor_name = "L8" if sensor_name == "Landsat 8" else "S2"
```
This can be taken from `CropParse.sensor` with the file name from each GeoJSON Feature's `properties.source`.

mcovalt commented on django/src/rdwatch/management/commands/ingest.py:
```diff
@@ -0,0 +1,272 @@
+            sensor_name = "L8" if sensor_name == "Landsat 8" else "S2"
+
+            satellite_timestamp = properties.get(
+                "observation_date", properties.get("start_date")
```
This can be taken from `CropParse.timestamp` with the file name from each GeoJSON Feature's `properties.source`. It does include date and time information.

mcovalt commented on django/src/rdwatch/management/commands/ingest.py:
```diff
@@ -0,0 +1,272 @@
+                / f"{crop_string}_salient.tif"
+            )
+
+            # TODO: not sure why some crop strings don't have
+            # a corresponding saliency TIF; there's quite a few in
+            # the sample data that don't. If one is encountered,
+            # just skip it and move on to the next raster.
```
Awesome, thanks for spotting this. Either:
1. Our view of the data model is incorrect (each `Site` must be derived from a `Saliency`)
2. We're missing data

I'm banking on number 2, but I'll consult the WATCH team to make sure. For now, logging this is and continuing on the error like you've done is perfect.

mcovalt commented on django/src/rdwatch/models/site.py:
```diff
@@ -15,6 +19,8 @@ class Site(models.Model):
     )
     saliency = models.ForeignKey(
         to="Saliency",
+        null=True,
+        blank=True,
```
Is this related to the [note here](https://github.com/ResonantGeoData/RD-WATCH/blob/98b94a11303be53937a417e14e3e780fc2f79789/django/src/rdwatch/management/commands/ingest.py#L176-L179)? Until we know for sure what's going on there, we should assume we're missing data rather than loosening our constraints. Right now, I can't mentally reconcile how a `Site` not derived from a `Saliency` can exist and our data model should reflect that.

mcovalt commented on django/src/rdwatch/management/commands/ingest.py:
```diff
@@ -0,0 +1,258 @@
+        sensor_name = crop_parse.sensor
+
+        assert sensor_name
+        sensor_name = "L8" if sensor_name == "Landsat 8" else "S2"
```
I think `crop_parse.sensor` will already be `"L8"` or `"S2"`, not `"Landsat 8"` or `"Sentinel 2"`.

mvandenburgh commented:
@mcovalt I removed the dependency on `django-click` for this script and just used the native Django mechanism for creating `manage.py` scripts. The syntax isn't as clean, but mypy was reporting errors with `django-click` and I figured it's not worth spending time debugging it if it's going to be somewhat temporary anyways. I also fixed the remaining linting errors, so this should be good to go unless you have further comments.

mcovalt reviewed (approved):
Looks good! I've put a _bunch_ of work on a branch from this, so let's go ahead and get this merged in. After playing with this ingest and the data, I've changed the data model quite a bit and thus the ingestion.
